src/sync/sync_config.hpp

```diff
@@ -110,14 +112,40 @@ struct SyncConfig {
     util::Optional<std::string> ssl_trust_certificate_path;
     std::function<sync::Session::SSLVerifyCallback> ssl_verify_callback;
     bool validate_sync_history = true;
+    bool is_partial = false;
+    util::Optional<std::string> custom_partial_sync_identifier;
+
+    std::string resolved_realm_url() const
```

Comment:
I'm a little torn on this. There are multiple aspects of the Realm URL that could, in theory, be resolved ('~` -> user ID, and partial sync identifier), but this only handles one aspect of it.

It's also not clear to me in what circumstances we should be calling this function vs accessing `realm_url` directly.

Reply:
The naming needs work, but I figured that it would be better to "construct" the final Realm URL intended to be passed into sync in one place, rather than having partial sync URLs be constructed ad-hoc in multiple places.

If it were up to me accessing `realm_url` directly would be forbidden, but it didn't seem practical to make that drastic a change.